### ISSUE TITLE
 Apache2 Ubuntu Default Page showing

### ISSUE THREAD

**Issue by a-i-m-a-r** [jschildgen/reveal.js-poll-plugin#1]
Hi Johannes,
I am willing to try your plugin but I am sorry I must ask this dummy question.
Initially, the server will be my own desktop.
I followed your instructions, but when I "Test the mobile website", it always shows the " Apache2 Ubuntu Default Page ".
When testing the API, I get "No method" and "{"OK":true}" as required.
How can I redirect to index.php (I suppose this is the target)?
BTW, where should I insert the IP of my desktop in proxy.php? 
Is it correct "echo file_get_contents("https://IP_ADDRESS/api/?" . $_SERVER['QUERY_STRING']);"?
Thank you.
aimar

**Comment by a-i-m-a-r** [comment]
Hi Johannes,
I solved the issue. I think it required a reboot so it could realize that I had removed the index.html file showing "Apache2 Ubuntu Default Page".
I am able to test the server as you indicated and everything shows accordingly.
However, if I launch a reveal.js presentation, the host sees the poll, but the rest get a "currently no poll running" message.
I think then that maybe I filled the proxy.php incorrectly and that is why the command to start the poll is not received by the server. Am I wrong?
Thank you.

**Comment by a-i-m-a-r** [comment]
Some more info. When reached the slide with the poll. the console of the browser give some error messages:

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at file://PATH_TO_POLL_FOLDER/poll/proxy.php/?method=start_poll&data=%7B%22question%22%3A%22Zein%20da%20galdera%3F%22%2C%22answers%22%3A%5B%22Galdera%20Okerra%22%2C%22Galdera%20Okerra%22%2C%22Galdera%20Okerra%22%2C%22Galdera%20Zuzena%22%5D%2C%22correct_answers%22%3A%5B%22Galdera%20Zuzena%22%5D%7D. (Reason: CORS request not http).

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at file://PATH_TO_POLL_FOLDER/poll/proxy.php/?method=status. (Reason: CORS request not http). 

I had similar messages for custom menu panels of the menu plugin for reveal.js. These issues are avoided when I upload the presentation to the Learning Management System of our University and launch the presentation from there instead of my computer.
But with the poll, the console says:
GET |PATH_TO_POLL_FOLDER_ON_THE_LMS/poll/proxy.php/?method=status
[HTTP/1.1 404 Not Found 137ms]
Any idea of can I prevent this issue when presenting from my computer?
Thank you.

**Comment by jschildgen** [comment]
Hi Aimar, you need a server that is accessible from the internet. It is not possible to run the poll system locally.

**Comment by jschildgen** [comment]
GET |PATH_TO_POLL_FOLDER_ON_THE_LMS/poll/proxy.php/?method=status

Remove the slash before the ?

**Comment by jschildgen** [comment]
Cross-Origin Request Blocked

For this reason, we use the proxy.php
It forwards each request to the server.

Mind that you also need a web server on your desktop! file://PATH_TO_POLL_FOLDER/poll/proxy.php will not work. You need Apache or Lighttpd. Then you access it via http://localhost/.../poll/proxy.php

**Comment by a-i-m-a-r** [comment]
Hi Johannes,
Thank you for your answers. I followed the instructions of the README file and I did install apache in the desktop. Now I am at home and can connect to it, but still get "currently no poll running", even though I remotely have launched the presentation in the desktop at work. I guess that I can not overcome the CORB if I launch it this way.
That is why I tried launching it from the LMS, but without luck.

> GET |PATH_TO_POLL_FOLDER_ON_THE_LMS/poll/proxy.php/?method=status
> Remove the slash before the ?

That is the response of the browser when I checked the console. I did not write that command, it was this way already in poll.js file. 
As you can guess I am not a developer and lack some necessary skills. May it be that I did not configure properly proxy.php?
Thank you again.

**Comment by jschildgen** [comment]
So, when your proxy.php content is echo file_get_contents("https://IP_ADDRESS/api/?" . $_SERVER['QUERY_STRING']);
Then please open https://IP_ADDRESS/api/ in a browser. Does it show No method?

Do you open your presentation via http://localhost You must not open it via file://...

**Comment by a-i-m-a-r** [comment]
Right. https://IP_ADDRESS/api/ leads to "No method"
I put the api folder within /var/www as suggested in the apache manual.
But I launch the presentation just clicking on the file in the file browser. So via file://... I guess.
If I need to open it via localhost, should I put it in /var/www/ also?

**Comment by a-i-m-a-r** [comment]
However, when I launch it on the LMS it is https://egela.ehu.eus/pluginfile.php/4723946/mod_resource/content/3/Adibidea.html
So, it is not locally via file://, but via https://.

**Comment by jschildgen** [comment]
Please write me an email then we can discuss this offline.

**Comment by a-i-m-a-r** [comment]
As you said, I had to launch the presentation, not clicking on the file (file://) but reaching it through the browser via http://
My error also was that I put the "poll" folder within the "plugin" folder, but it must remain along the presentation main file.
But I had to change the question text colour and the number of responses text colour (otherwise it was white, blending with the white of the box and making it invisible).
Thank you for your help.
I will proceed to close this issue.
